ext/intl/formatter/formatter_format.c

```diff
@@ -76,7 +100,13 @@ PHP_FUNCTION( numfmt_format )
 
 		case FORMAT_TYPE_INT64:
 		{
-			int64_t value = (Z_TYPE_P(number) == IS_DOUBLE)?(int64_t)Z_DVAL_P(number):Z_LVAL_P(number);
+			int64_t value;
+			if (Z_TYPE_P(number) == IS_DOUBLE) {
+				value = (int64_t)Z_DVAL_P(number);
+			} else {
+				convert_to_long(number);
+				value = Z_LVAL_P(number);
```

Comment:
This will cast non-numeric strings to 0, which is not what should happen.

Same as the other cases using ``convert_to_long()`` or ``convert_to_double()``

Reply:
Okay, what should happen instead?

Reply:
Or to put the question slightly differently: I’m not very familiar with the php-src codebase, and couldn’t even find documentation on what exactly `convert_to_long()` does – if there’s another paradigm for what the code should do with non-numeric strings, such as throwing an error, can you point me to another place in PHP that uses this paradigm, so I can see how it’s implemented there?

Reply:
I switched it to `zval_try_get_long()` (in a separate commit), but it seems to introduce a lot of deprecation warnings that are currently causing tests to fail. Should I add the deprecations to the expected test output, or should the deprecations not happen after all?

Also added a `throw` for cases where the `INT32` type is used with numbers that exceed 32 bits of precision, but I’m not sure whether that’s what you had in mind or not. Happy to change this part.

`convert_to_double()` not converted, because I couldn’t find a `zval_try_get_double()` or equivalent function.